packages/open-api-gateway/src/project/codegen/generated-java-client-project.ts

```diff
@@ -83,6 +84,19 @@ export class GeneratedJavaClientProject extends JavaProject {
       specPath: options.specPath,
       invokeGenerator: options.generateClient,
     });
+
+    // Use a package.json to ensure the client is discoverable by nx
+    new JsonFile(this, "package.json", {
+      obj: {
+        name: this.name,
+        __pdk__: true,
+        version: options.version,
+        scripts: Object.fromEntries(
+          this.tasks.all.map((task) => [task.name, `npx projen ${task.name}`])
+        ),
+      },
+      readonly: true,
```

Comment:
should we also set the marker given this is generated?

Reply:
same for all other langs.

Reply:
I think given that all the rest of the generated client code won't have the marker there's not much benefit in adding it for the `package.json`. Greg is actually looking into adding more general cleanup of client projects - ie when you remove a `ClientLanguage` it deletes the appropriate directory, or when you change the spec you don't end up with old dangling generated files :)

Reply:
Ok sounds good.